flow.c

```diff
@@ -1830,11 +1856,19 @@ void* Flow_Worker(void* User)
 								for (int i=0; i < SortListCnt[j]; i++)
 								{
 									u32 FIndex =  SortList[j][i];
-									FlowRecord_t* Flow = &FlowIndex->FlowList[ FIndex ];	
-
-									JSONBufferOffset += FlowDump(JSONBuffer + JSONBufferOffset, PktBlock->TSSnapshot, Flow, FlowCnt, FlowTotal);
+									FlowRecord_t* Flow = &FlowIndex->FlowList[ FIndex ];
+									// TSFirst will be 0 when we're flushing the
+									// last packets from Flow_Close, since the
+									// main loop didn't set TSFirst
+									u64 PacketTS = PktBlock->TSFirst == 0 ? PktBlock->TSSnapshot : PktBlock->TSFirst;
+									u64 FlowTS = (u64) (PacketTS / g_FlowSampleRate) * g_FlowSampleRate;
+									// Can't have more events than packets in the flow record
+									assert(Flow->TotalPkt >= Flow->TCPEventCount);
+									JSONBufferOffset += FlowDump(JSONBuffer + JSONBufferOffset, FlowTS, Flow, FlowCnt, FlowTotal);
```

Comment:
@fmadio Wanted to confirm that using `TSFirst` instead of `TSSnapshot` for all `FlowRecord_t` (and `TCPEvent_t`) is alright (and won't have unintended consequences)